### PR TITLE
[CFBG module part 3] Core/BG: Rework functions, delete ASSERT's.

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
Continuation #2008
Continuation #2009

##### CHANGES PROPOSED:
- Rework functions, added new args, delete ASSERT's

##### TESTS PERFORMED:
- Build in win10 (VS 16.1.3)
- In game

##### HOW TO TEST THE CHANGES:
* Need to test all queue
* * Reg all BG and enters battle
* * Reg random BG and enter battle
* * Reg arena rating and no rating

##### Target branch(es): **Master**

### CODE REVIEW

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 1]
```diff
@@ -155,11 +158,8 @@ GroupQueueInfo* BattlegroundQueue::AddGroup(Player * leader, Group * grp, PvPDif
 
     // announce world (this doesn't need mutex)
     if (isRated && sWorld->getBoolConfig(CONFIG_ARENA_QUEUE_ANNOUNCER_ENABLE))
-    {
-        ArenaTeam* team = sArenaTeamMgr->GetArenaTeamById(arenateamid);
-        if (team)
+        if (ArenaTeam * team = sArenaTeamMgr->GetArenaTeamById(arenateamid))
             sWorld->SendWorldText(LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN, team->GetName().c_str(), ginfo->ArenaType, ginfo->ArenaType, ginfo->ArenaTeamRating);
-    }
```
Comment (Viste) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: lol

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 2]
```diff
@@ -155,11 +158,8 @@ GroupQueueInfo* BattlegroundQueue::AddGroup(Player * leader, Group * grp, PvPDif
 
     // announce world (this doesn't need mutex)
     if (isRated && sWorld->getBoolConfig(CONFIG_ARENA_QUEUE_ANNOUNCER_ENABLE))
-    {
-        ArenaTeam* team = sArenaTeamMgr->GetArenaTeamById(arenateamid);
-        if (team)
+        if (ArenaTeam * team = sArenaTeamMgr->GetArenaTeamById(arenateamid))
             sWorld->SendWorldText(LANG_ARENA_QUEUE_ANNOUNCE_WORLD_JOIN, team->GetName().c_str(), ginfo->ArenaType, ginfo->ArenaType, ginfo->ArenaTeamRating);
```
Comment (Viste) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: lol

Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: :D

Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: maybe rework AC BG ? :D

**Comment by Winfidonarleyan** [comment]
I guess no one wants the module :/

**Comment by spinzaku** [comment]
i will test tomorrow!

**Comment by Winfidonarleyan** [comment]
@spinzaku Thanks 👍

File: src/server/game/Battlegrounds/Battleground.h [hunk 3]
```diff
@@ -396,7 +396,7 @@ class Battleground
         void SetMaxPlayersPerTeam(uint32 MaxPlayers) { m_MaxPlayersPerTeam = MaxPlayers; }
         void SetMinPlayersPerTeam(uint32 MinPlayers) { m_MinPlayersPerTeam = MinPlayers; }
 
-        void DecreaseInvitedCount(TeamId teamId)    { ASSERT(m_BgInvitedPlayers[teamId] > 0); --m_BgInvitedPlayers[teamId]; }
+        void DecreaseInvitedCount(TeamId teamId)    { if (m_BgInvitedPlayers[teamId]) --m_BgInvitedPlayers[teamId]; }
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/Battleground.h]: Do you understand why an ASSERT was used here?

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp [hunk 4]
```diff
@@ -923,7 +923,8 @@ void BattlegroundMgr::RemoveBattleground(BattlegroundTypeId bgTypeId, uint32 ins
 
 void BattlegroundMgr::InviteGroupToBG(GroupQueueInfo* ginfo, Battleground* bg, TeamId teamId)
 {
-    ASSERT(!ginfo->IsInvitedToBGInstanceGUID);
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundMgr.cpp]: Do you understand why an ASSERT was used here?

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp [hunk 5]
```diff
@@ -942,13 +943,12 @@ void BattlegroundMgr::InviteGroupToBG(GroupQueueInfo* ginfo, Battleground* bg, T
...
 
     // loop through the players
-    for (std::set<uint64>::iterator itr = ginfo->Players.begin(); itr != ginfo->Players.end(); ++itr)
+    for (auto itr : ginfo->Players)
     {
         // get the player
-        Player* player = ObjectAccessor::FindPlayerInOrOutOfWorld(*itr);
-
-        // player is removed from queue when logging out
-        ASSERT(player);
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundMgr.cpp]: Do you understand why an ASSERT was used here?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 6]
```diff
@@ -463,8 +456,10 @@ void BattlegroundQueue::FillPlayersToBG(const int32 aliFree, const int32 hordeFr
     }
 }
 
-void BattlegroundQueue::FillPlayersToBGWithSpecific(const int32 aliFree, const int32 hordeFree, BattlegroundBracketId thisBracketId, BattlegroundQueue * specificQueue, BattlegroundBracketId specificBracketId)
+void BattlegroundQueue::FillPlayersToBGWithSpecific(Battleground* bg, const int32 aliFree, const int32 hordeFree, BattlegroundBracketId thisBracketId, BattlegroundQueue * specificQueue, BattlegroundBracketId specificBracketId)
 {
+    UNUSED(bg);
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Unless I'm missing something, it doesn't make sense to pass a Battleground* here?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 7]
```diff
@@ -407,8 +398,10 @@ bool BattlegroundQueue::GetPlayerGroupInfoData(uint64 guid, GroupQueueInfo * gin
 }
 
 // this function is filling pools given free slots on both sides, result is ballanced
-void BattlegroundQueue::FillPlayersToBG(const int32 aliFree, const int32 hordeFree, BattlegroundBracketId bracket_id)
+void BattlegroundQueue::FillPlayersToBG(Battleground* bg, const int32 aliFree, const int32 hordeFree, BattlegroundBracketId bracket_id)
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Unless I'm missing something, it doesn't make sense to pass a Battleground* here?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 8]
```diff
@@ -963,6 +955,31 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
     }
 }
 
+uint32 BattlegroundQueue::GetPlayesCountInGroupsQueue(BattlegroundBracketId bracketId, BattlegroundQueueGroupTypes bgqueue)
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: GetPlayerCountInGroupsQueue?

File: src/server/game/Battlegrounds/BattlegroundQueue.h [hunk 9]
```diff
@@ -68,6 +70,8 @@ class BattlegroundQueue
         bool GetPlayerGroupInfoData(uint64 guid, GroupQueueInfo* ginfo);
         void PlayerInvitedToBGUpdateAverageWaitTime(GroupQueueInfo* ginfo);
         uint32 GetAverageQueueWaitTime(GroupQueueInfo* ginfo) const;
+        uint32 GetPlayesCountInGroupsQueue(BattlegroundBracketId bracketId, BattlegroundQueueGroupTypes bgqueue);
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundQueue.h]: GetPlayerCountInGroupsQueue?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 10]
```diff
@@ -963,6 +955,31 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
...
+{
+    uint32 PlayesCount = 0;
+
+    for (auto itr : m_QueuedGroups[bracketId][bgqueue])
+        if (!itr->IsInvitedToBGInstanceGUID)
+            PlayesCount += (uint32)itr->Players.size();
```
Comment (NixAJ) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: perhaps a static_cast instead?

File: src/server/game/Battlegrounds/Battleground.h (hunk 3 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/Battleground.h]: No

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp (hunk 4 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp]: No

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp (hunk 5 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp]: No

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 6 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: For normal support module

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 7 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: For normal support module

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 8 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Yes

File: src/server/game/Battlegrounds/BattlegroundQueue.h (hunk 9 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.h]: Yes

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 10 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Maybe

File: src/server/game/Battlegrounds/Battleground.h (hunk 3 above)
Comment (Viste) [reply src/server/game/Battlegrounds/Battleground.h]: for zero check?

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp (hunk 4 above)
Comment (Viste) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp]: for zero check?

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp (hunk 5 above)
Comment (Viste) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp]: for zero check?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 8 above)
Comment (Stoabrogga) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Had to read this twice... There's a typo in there:
GetPlaye**s**CountInGroupsQueue
Should be renamed to
GetPlaye**r**CountInGroupsQueue

File: src/server/game/Battlegrounds/BattlegroundQueue.h (hunk 9 above)
Comment (Stoabrogga) [reply src/server/game/Battlegrounds/BattlegroundQueue.h]: Typo:
GetPlaye**s**CountInGroupsQueue
Should be renamed to
GetPlaye**r**CountInGroupsQueue

**Comment by DARKENE** [comment]
Crossfaction battlegrounds is very hard, need to Professional Coder

I think Only iranian Can To fixed for Azerothcore 3.3.5, Ask for help!

https://gist.github.com/irancore/10913800#gistcomment-2200227

**Comment by poszer** [comment]
> Crossfaction battlegrounds is very hard, need to Professional Coder
> 
> I think Only iranian Can To fixed for Azerothcore 3.3.5, Ask for help!
> 
> https://gist.github.com/irancore/10913800#gistcomment-2200227

It's already fixed

**Comment by DARKENE** [comment]
why ! how to use ! NEED PATCH?

Ican use?

**Comment by Winfidonarleyan** [comment]
https://github.com/Winfidonarleyan/mod-cfbg
The module is ready, you just need to prepare the core

**Comment by DARKENE** [comment]
Hi @Winfidonarleyan 
 how to prepare the core ! i need patch?

**Comment by Winfidonarleyan** [comment]
Need test this pr

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 11]
```diff
@@ -963,6 +955,31 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
     }
 }
 
+uint32 BattlegroundQueue::GetPlayesCountInGroupsQueue(BattlegroundBracketId bracketId, BattlegroundQueueGroupTypes bgqueue)
+{
+    uint32 PlayesCount = 0;
```
Comment (P-Kito) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Lower case variable and another Typo in there PlayeR not PlayeS

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 12]
```diff
@@ -963,6 +955,31 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
...
+
+    for (auto itr : m_QueuedGroups[bracketId][bgqueue])
```
Comment (P-Kito) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Why is this a for-loop btw? Arent you accessing a single entry by bracketId and bgQueue?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 13]
```diff
@@ -963,6 +955,31 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
...
+        if (!itr->IsInvitedToBGInstanceGUID)
+            PlayesCount += (uint32)itr->Players.size();
+
+    return PlayesCount;
+}
+
+bool BattlegroundQueue::IsEmptyAllQueue(BattlegroundBracketId bracket_id)
+{
+    uint32 QueueEmptyCount = 0;
```
Comment (P-Kito) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: lower case variable name here

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 14]
```diff
@@ -963,6 +955,31 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
...
+    uint32 PlayesCount = 0;
+
+    for (auto itr : m_QueuedGroups[bracketId][bgqueue])
+        if (!itr->IsInvitedToBGInstanceGUID)
+            PlayesCount += (uint32)itr->Players.size();
+
+    return PlayesCount;
+}
+
+bool BattlegroundQueue::IsEmptyAllQueue(BattlegroundBracketId bracket_id)
```
Comment (P-Kito) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: I suggest:    AllQueuesEmpty() or anything else... but "IsEmptyAllQueue" sounds weird.

Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: maybe `IsAllQueuesEmpty()` ?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 13 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Okay

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 12 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: Ah, yes yes

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp (hunk 11 above)
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: My bad, fix

**Comment by Winfidonarleyan** [comment]
@P-Kito thanks for feedback

**Comment by solidmaxtor** [comment]
Have it running This PR 3 days on a server with more 300 on line.
The queues of bg and arenas work normally, without any crahs or issues sucess.

**Comment by Winfidonarleyan** [comment]
Please retest with new commits

**Comment by Winfidonarleyan** [comment]
Ah, I'll fix it later, I don't have a laptop right now.

**Comment by Winfidonarleyan** [comment]
So, I fix, need a retest

**Comment by solidmaxtor** [comment]
Work fine for me,  no reports o crashes.

**Comment by Winfidonarleyan** [comment]
Good, thanks for test @solidmaxtor

**Comment by xDevICCI** [comment]
Hi guys, this error ! :/
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:352:82: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...&& bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].empty() && spec...
                                               ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:352:157: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...&& specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].empty())
                                                         ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:357:124: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].begin()...
                                                      ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:357:197: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].end());
                                                      ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:358:114: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].begin(), bgqueue->...
                                            ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:358:177: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].end());
                                            ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:359:118: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].begin()...
                                                      ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:359:191: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].end());
                                                      ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:360:108: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].begin(), bgqueue->...
                                            ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:360:171: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].end());
                                            ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:419:73: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
        if ((aliFree > hordeFree && bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE...
                                                                        ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:423:102: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...Ali_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
                                                   ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:424:59: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
    for (; Ali_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG]...
                                                          ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:427:104: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...Horde_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
                                                     ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:428:61: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
    for (; Horde_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFB...
                                                            ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:446:71: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...for (; Ali_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].end...
                                                           ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:456:73: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
  ...for (; Horde_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].e...
                                                             ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:496:32: error: 
      no member named 'GetPlayersCountInGroupsQueue' in 'BattlegroundQueue'
    uint32 qPlayers = bgqueue->GetPlayersCountInGroupsQueue(bracketId, B...
                      ~~~~~~~  ^
/Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:496:72: error: 
      use of undeclared identifier 'BG_QUEUE_CFBG'
    uint32 qPlayers = bgqueue->GetPlayersCountInGroupsQueue(bracketId, BG_QU...
                                                                       ^
1 warning generated.
19 errors generated.
make[2]: *** [src/server/scripts/CMakeFiles/scripts.dir/__/__/__/modules/mod-cfbg/src/CFBG.cpp.o] Error 1

**Comment by WizardsBR** [comment]
> Hi guys, this error ! :/
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:352:82: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...&& bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].empty() && spec...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:352:157: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...&& specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].empty())
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:357:124: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].begin()...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:357:197: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].end());
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:358:114: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].begin(), bgqueue->...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:358:177: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].end());
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:359:118: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].begin()...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:359:191: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...specificQueue->m_QueuedGroups[specificBracketId][BG_QUEUE_CFBG].end());
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:360:108: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].begin(), bgqueue->...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:360:171: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...bgqueue->m_QueuedGroups[thisBracketId][BG_QUEUE_CFBG].end());
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:419:73: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> if ((aliFree > hordeFree && bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:423:102: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...Ali_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:424:59: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> for (; Ali_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG]...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:427:104: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...Horde_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:428:61: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> for (; Horde_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFB...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:446:71: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...for (; Ali_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].end...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:456:73: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> ...for (; Horde_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].e...
> ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:496:32: error:
> no member named 'GetPlayersCountInGroupsQueue' in 'BattlegroundQueue'
> uint32 qPlayers = bgqueue->GetPlayersCountInGroupsQueue(bracketId, B...
> ~~~~~~~ ^
> /Users/victorgodoyroa/desktop/chilewow/modules/mod-cfbg/src/CFBG.cpp:496:72: error:
> use of undeclared identifier 'BG_QUEUE_CFBG'
> uint32 qPlayers = bgqueue->GetPlayersCountInGroupsQueue(bracketId, BG_QU...
> ^
> 1 warning generated.
> 19 errors generated.
> make[2]: *** [src/server/scripts/CMakeFiles/scripts.dir/**/**/__/modules/mod-cfbg/src/CFBG.cpp.o] Error 1

add the commits

https://github.com/azerothcore/azerothcore-wotlk/pull/2064/commits

**Comment by Winfidonarleyan** [comment]
@xDevICCI Need apply PR #2064

**Pull request merged**